go-controller/pkg/ovn/ovn.go

```diff
@@ -332,33 +332,34 @@ func podScheduledAndWantsNetwork(pod *kapi.Pod) bool {
 
 // WatchPods starts the watching of Pod resource and calls back the appropriate handler logic
 func (oc *Controller) WatchPods() error {
-	handledPods := sets.String{}
+	retryPods := sets.String{}
 	_, err := oc.watchFactory.AddPodHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			pod := obj.(*kapi.Pod)
-			if !podScheduledAndWantsNetwork(pod) {
-				return
-			}
-			if err := oc.addLogicalPort(pod); err != nil {
-				logrus.Errorf(err.Error())
+			if podScheduledAndWantsNetwork(pod) {
+				if err := oc.addLogicalPort(pod); err != nil {
+					logrus.Errorf(err.Error())
+					retryPods.Insert(string(pod.UID))
+				}
 			} else {
-				handledPods.Insert(string(pod.UID))
+				// Handled unscheduled pods later on Update
+				retryPods.Insert(string(pod.UID))
```

Comment:
@dcbw was about to comment on this, but your latest changes captures it :-).

Only caveat here is that the retryPods now will also have pods with HostNetwork, set?. Perhaps @L346 we need to wrap the Insert with `if !pod.spec.Hostnetwork`?